large-anon/src/Data/Record/Anon/Internal/Advanced.hs

```diff
@@ -341,6 +343,24 @@ sequenceA' = sequenceA . co
     co :: Record m r -> Record (m :.: I) r
     co = noInlineUnsafeCo
 
+distribute :: forall r m f.
+     (Functor m, KnownFields r)
+  => m (Record f r) -> Record (m :.: f) r
+distribute =
+      unsafeFromCanonical
+    . Canon.fromList
+    . (\cs -> fieldVec cs <$> indices)
+    . fmap toCanonical
+  where
+    indices :: [Int]
+    indices = [0 .. pred (length $ proxy fieldNames (Proxy @r))]
+
+    fieldVec :: m (Canonical f) -> Int -> (m :.: f) Any
+    fieldVec cs idx = Comp $ flip Canon.getAtIndex idx <$> cs
+
+distribute' :: (Functor m, KnownFields r) => m (Record I r) -> Record m r
+distribute' = map (fmap unI . unComp) . distribute
```

Comment:
I am never sure about the performance implications, but what would you think about coercing similar to e.g. `sequenceA'` ?

Reply:
Yes, fair enough, have changed it.